test/Microsoft.ApplicationInsights.AspNetCore.Tests/Helpers/HttpContextAccessorHelper.cs

```diff
@@ -22,18 +22,18 @@ public static HttpContextAccessor CreateHttpContextAccessor(RequestTelemetry req
             request.Path = new PathString("/Test");
             var contextAccessor = new HttpContextAccessor() { HttpContext = request.HttpContext };
 
-            services.AddInstance<IHttpContextAccessor>(contextAccessor);
+            services.AddSingleton<IHttpContextAccessor>(contextAccessor);
 
             if (actionContext != null)
             {
                 var si = new ActionContextAccessor();
                 si.ActionContext = actionContext;
-                services.AddInstance<IActionContextAccessor>(si);
+                services.AddSingleton<IActionContextAccessor>(si);
```

Comment:
It didn't really matter here, as we update the context on every test. We update the service associated with request telemetry in the context on every run, which as a result will not effect if the service is singleton or scoped.